plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FutureNonSystemJobSuspender.java

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.IJobChangeListener;
+import org.eclipse.core.runtime.jobs.Job;
+
+/**
+ * Prevents scheduling all future non-system jobs once {@link #suspendFutureJobs} is called, until
+ * {@link #resume} is called. Jobs already scheduled are not affected and will run to completion.
+ *
+ * The class is for https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155. The
+ * purpose of the class is to prevent the second ConvertJob from running. The second ConvertJob is
+ * triggered by the first ConvertJob when the latter job installs the JSDT facet.
+ *
+ * Not recommended to use for other situations, although the workings of the class are general.
+ */
+public class FutureNonSystemJobSuspender {
+  private static boolean suspended;
+
+  private static class SuspendedJob {
+    private Job job;
+    private long scheduleDelay;  // ms
+
+    private SuspendedJob(Job job, long scheduleDelay) {
+      this.job = job;
+      this.scheduleDelay = scheduleDelay;
+    }
+  }
+
+  private static List<SuspendedJob> suspendedJobs =
+      Collections.synchronizedList(new ArrayList<SuspendedJob>());
+
+  private static JobScheduleListener jobScheduleListener = new JobScheduleListener();
+
+  /** Once called, it is imperative to call {@link resume()} later. */
+  public static synchronized void suspendFutureJobs() {
+    Preconditions.checkState(!suspended, "Already suspended.");
+    suspended = true;
+    Job.getJobManager().addJobChangeListener(jobScheduleListener);
+  }
+
+  public static synchronized void resume() {
+    Preconditions.checkState(suspended, "Not suspended.");
+    resumeInternal();
+  }
+
+  @VisibleForTesting
+  static synchronized void resumeInternal() {
+    suspended = false;
+    Job.getJobManager().removeJobChangeListener(jobScheduleListener);
+
+    for (SuspendedJob jobInfo : suspendedJobs) {
+      jobInfo.job.schedule(jobInfo.scheduleDelay);
+    }
+    suspendedJobs.clear();
+  }
+
+  private FutureNonSystemJobSuspender() {}
+
+  /** Listens for every job being scheduled and cancels it. */
+  private static class JobScheduleListener implements IJobChangeListener {
+    @Override
+    public void scheduled(IJobChangeEvent event) {
+      Job job = event.getJob();
+      if (!job.isSystem()) {
+        job.cancel();  // This will always succeed since the job is not running yet.
+        suspendedJobs.add(new SuspendedJob(job, event.getDelay()));
```

Comment:
This doesn't seem to share the same lock as the synchronized methods. Is this a problem? I.e. jobs can be cancelled and added in the middle of a synchronized method. I suggest:

1. Extract these inner classes to outer classes.
2. Add a synchronized add() method to FutureNonSystemJobSuspender that this method calls instead of adding directly into suspendedJobs.